mta/base/application/implementations/operator_ui.py

```diff
@@ -0,0 +1,288 @@
+            return page_safe
+
+        wait_for(_check, timeout=timeout, delay=0.2, silent_failure=True, very_quiet=True)
+
+    def after_keyboard_input(self, element, keyboard_input):
+        self.browser.plugin.ensure_page_safe()
+
+    def after_click(self, element, locator):
+        # page_dirty is set to None because otherwise if it was true, all next ensure_page_safe
+        # calls would check alert presence which is enormously slow in selenium.
+        self.browser.page_dirty = None
+
+
+class MTABrowser(Browser):
+    def __init__(self, selenium_webdriver, endpoint, extra_objects=None):
+        extra_objects = extra_objects or {}
+        extra_objects.update({"application": endpoint.owner, "endpoint": endpoint})
+        super(MTABrowser, self).__init__(
+            selenium_webdriver,
+            plugin_class=MTABrowserPlugin,
+            logger=logger,
+            extra_objects=extra_objects,
+        )
+        self.window_handle = selenium_webdriver.current_window_handle
+        self.logger.info(
+            "Opened browser %s %s",
+            selenium_webdriver.capabilities.get("browserName", "unknown"),
+            selenium_webdriver.capabilities.get("version", "unknown"),
+        )
+
+    @property
+    def application(self):
+        return self.extra_objects["application"]
+
+    def create_view(self, *args, **kwargs):
+        return self.application.operator_ui.create_view(*args, **kwargs)
+
+    @property
+    def product_version(self):
+        return self.application.version
+
+
+def can_skip_pre_nav_prep(fn):
+    """Decorator for setting a noop"""
+    fn._can_skip_pre_nav_prep = True
+    return fn
+
+
+class NavigatableMixin(object):
+    """NavigatableMixin ensures that an object can navigate properly
+
+    The NavigatableMixin object ensures that a Collection/Entity object inside the
+    framework has access to be able to **create** a Widgetastic View, and that it
+    has access to the browser.
+
+    Note: The browser access will have to change once proliferation of the Sentaku
+          system becomes common place
+    """
+
+    @property
+    def browser(self):
+        return self.application.operator_ui.widgetastic_browser
+
+    def create_view(self, view_class, o=None, override=None):
+        o = o or self
+        if override is not None:
+            new_obj = copy(o)
+            new_obj.__dict__.update(override)
+        else:
+            new_obj = o
+        return self.application.operator_ui.create_view(
+            view_class, additional_context={"object": new_obj}
+        )
+
+    def list_destinations(self):
+        """This function returns a list of all valid destinations for a particular object
+        """
+        return {
+            impl.name: impl.navigator.list_destinations(self)
+            for impl in self.application.context.implementations.values()
+            if impl.navigator
+        }
+
+
+class MTANavigateStep(NavigateStep):
```

Comment:
do we need two separate navigation steps?
I think we already covered it and better to have to use the same for both variant `WebUI` and `OperatorWebUI`